src/semantic/type_checker.rs

```diff
@@ -430,15 +413,20 @@ impl Type {
                 self.occurs(*t2)
             }
             ClassType {
-                type_parameters, ..
-            } => {
-                for t in type_parameters {
-                    if self.occurs(t) {
-                        return true;
+                name,
+                type_parameters,
+                ..
+            } => match name.as_str() {
+                "void" | "int" | "bool" | "f64" | "string" => false,
```

Comment:
These types still won't be occurred? @YanHerChen NOTE me that should be changed to if `type_parameters.len() == 0`, this is a general and correct check.